server/src/main/java/org/elasticsearch/search/aggregations/metrics/CardinalityAggregator.java

```diff
@@ -193,13 +216,29 @@ public void close() {
         }
     }
 
-    private static class DirectCollector extends Collector {
-
-        private final MurmurHash3Values hashes;
-        private final HyperLogLogPlusPlus counts;
+    private abstract static class DirectCollector extends Collector {
+        protected final HyperLogLogPlusPlus counts;
 
-        DirectCollector(HyperLogLogPlusPlus counts, MurmurHash3Values values) {
+        DirectCollector(HyperLogLogPlusPlus counts) {
             this.counts = counts;
+        }
+
+        @Override
+        public void postCollect() {
+            // no-op
+        }
+
+        @Override
+        public void close() {
+            // no-op
```

Comment:
Do we not need to release the HLL++ here? If it is indeed correct to not release it, a comment as to why would be great.